desktop docs: initial draft (#3571)
chef/automate: components/automate-chef-io/content/docs/desktop.md

+++
title = "Desktop"
description = "Using the Desktop View"
date = 2018-03-26T16:01:47-07:00

> draft = true

Reviewer: NTS: FIX before merging!

> draft = false

Reviewer 2: ```suggestion
draft = true
```

We don't want to publish these until June

bref = ""
toc = true
+++

## Overview
The _Desktop_ page displays information about all desktops connected to Chef Automate. It can be found via the top nav of your Automate instance, at `automate-url/desktop`
Desktops appear in this view after a Chef Infra Client run has executed.

### Daily Check In
Shows a count of desktops that have and are reporting in to Automate. Desktops that have not checked in in the last twenty-four hours will be counted as `unknown`.
Selecting a status filter opens the list view of the desktops (Insight) with the selected status.

### Check In History
Shows the history of checked-in and unknown desktop counts over time. Timescale can be adjusted using the timescale dropdown menu.

### Top Errors
Shows the most common errors that have occurred across all desktops over the last twenty-four hours, and a count of the machines that incurred the error.
Selecting an error opens the list view of the desktops (Insight) with the error to enable further investigation.

> ### Unknown Duration

Reviewer: @vjeffrey is this now "Time since Last Check-in"?

Author: ya!

Shows count of desktops that have been `unknown` (have not reported in to Automate) for the given time frames.

> ### Insight (Desktop List)

Reviewer: @vjeffrey I don't see this as a display/section on the Desktop Management Dashboard. Is this still in development or am I looking in the wrong place?

Reviewer 3: if you interact with one of the table rows in `Time since Last Check-in` it should open up
<img width="1333" alt="insights-panel" src="https://user-images.githubusercontent.com/5489125/83028840-cedb6280-9fe6-11ea-9bce-da9d52e2df74.png">

> ### Insight (Desktop List)

Reviewer: @vjeffrey Do we have a better name for this display? I  do understand we're trying to not label everything "Desktop."

Author: 🤷‍♀️ that's the name from the designs... @apriofrost any ideas?

Reviewer 3: if we arent calling this insight in the ui? then i dont think we should have that name here.. unless yall are planning to add that heading later

Reviewer: There's no label for this in the UI, but I do need something to describe it. "list view" or "summary list" have been floated about, but...

Author: "filtered desktop list" ?

Shows a list of desktops. Available filters found at the top of the list. Additional filtering attributes may be selected using the `Choose Attributes` dropdown menu.

> ### Node Detail

Reviewer: @vjeffrey I don't see this as a display/section on the Desktop Management Dashboard. Is this still in development or am I looking in the wrong place?

Author: same deal here as what susan said above: to get there you gave to click on a node in the insights list. 
so it goes dashboard -> select a thing -> list -> select a machine -> see details

Author: same deal here as what susan said above: to get there you gave to click on a node in the insights list. 
so it goes dashboard -> select a thing -> list -> select a machine -> see details

Shows details of a single desktop and a graph displaying the latest status per day for the desktop (failed, converged, or unknown).

### Configuration
The Automate install for the desktop experience requires the use of a flag, `--with-desktop`.

> When starting Automate with the desktop flag, data lifecycle configurations will be set to never mark nodes as missing and never delete missing nodes. Users are encouraged to not modify these settings.

Reviewer 3: whoa, i prolly dont know enough. but i feel like support has this push right now to get everyone to set these. are folks not going to run into space issues with these settings?

Reviewer 2: I'm just getting started on the Desktop docs. I'll confirm what our base case is.

Author: @susanev are you thinking about retention settings? this is slightly different -about nodes going missing/deleting missing nodes. the default config for that is to mark as missing after 1 day, i believe, and then delete if missing a week. this shouldn’t affect space negatively. the presumption is that desktops aren’t ephemeral, so the whole “missing” thing for a desktop is actually “unknown”

Author: @susanev are you thinking about retention settings? this is slightly different -about nodes going missing/deleting missing nodes. the default config for that is to mark as missing after 1 day, i believe, and then delete if missing a week. this shouldn’t affect space negatively. the presumption is that desktops aren’t ephemeral, so the whole “missing” thing for a desktop is actually “unknown”

Author: @susanev are you thinking about retention settings? this is slightly different -about nodes going missing/deleting missing nodes. the default config for that is to mark as missing after 1 day, i believe, and then delete if missing a week. this shouldn’t affect space negatively. the presumption is that desktops aren’t ephemeral, so the whole “missing” thing for a desktop is actually “unknown”

Reviewer 3: oh yah, totally was. thanks for clarifying.

Author: hmm, good idea. ill add it to the issue for all that stuff!

Author: flag is actually `--product desktop`

Reviewer 3: does this need more information on how to use the flag? or a link to the config page or something?

Reviewer: I was planning on that @susanev but the deploy command listed in https://automate.chef.io/docs/install/#deploy-chef-automate looked significantly different than what @lancewf had demo'd last Friday.

Author: the only difference to the deploy command usually used is adding `--product desktop` -- just like we do for bldr and chef server installs

Reviewer 3: we could just provide the exact command here, along with the patch command if they already have automate deployed if we didnt want to link to stuff. also totally fine to come back to this later if yall want 🤷 

ive been fielding what seems like a never-ending stream of how do i add a product questions in success slack so thought maybe we should be a bit more specific

### Other Useful Information
The desktop view does not support project filtering.

> No compliance profiles are supported for the Desktop view; they will not be included when using the desktop flag.

Author: we mention the profiles a bit here

Reviewer: @vjeffrey Are compliance profiles not included at all as a part of the `--product desktop` installation process, too?

Author: yup. no available profiles when installed with desktop flag.
users will still be able to upload their own profiles if they want, but nothing will be shipped with automate

> Liveness agents are not excluded from node counts in the desktop view.

Reviewer: @vjeffrey Are these Liveness agents considered general liveness agents or something specific to Automate? These are not documented anywhere in Automate currently.

Author: https://github.com/chef/automate-liveness-agent

@lancewf do you have more words you could add to help clarify?

Reviewer 4: The liveness agent is only used for the AWS OpsWorks. It is not specific to Automate. It is installed on the AWS nodes with Chef Infra Client. 
@danielsdeleo Might have more information.

+++
title = "Desktop Dashboard"
description = "Using the Desktop Dashboard"
date = 2018-03-26T16:01:47-07:00
draft = false
bref = ""
toc = true
+++

The Chef Automate _Desktop_ dashboard displays status information about all desktops connected to Chef Automate.
Desktop information populates this dashboard after a Chef Infra Client run has executed.

> Node counts in the _Desktop_ dashboard may include liveness agents. Desktop information appears in a series of displays. The _Desktop_ dashbaord does not support project filtering.

Reviewer: Will need to move this around.

## Setting Up The Desktop Dashboard

> Include a `--product desktop` flag to enable the Desktop dashboard when [installing Chef Automate](https://automate.chef.io/docs/infra-server/). The Desktop dashboard has no supported compliance profiles, and no compliance profiles will be included in installation when using the `--product desktop` flag.

Reviewer: @vjeffrey @susanev I used this link for installation reference per our previous comment thread.

Author: um, why are we linking to the automate chef server install? is it bc that install also uses the --product flag?

Reviewer 3: we should prolly link to config, but also i think @yzl has a pr to add a section to config about adding products.. 🤷

Reviewer 2: We should give the full command (maybe we do?) The command I was given  last week is: `chef-automate deploy --product automate --product infra-server --product desktop`. Is that correct?  The automate + infra install docs are the most comprehensive for system requirements and installation methods.

Author: we do not need the --product automate part. we only need the chef server part if they're also planning on doing the chefserver/automate on the same box, and not using a sep chef server

Reviewer: So then sending them to the different ways to install Chef Infra Server and/or Chef Automate together doc is a relevant link...?

Below the bar, boxes displays counts for desktops with `unknown` and `checked-in` statuses, and a total count of desktops.
`Checked-in` refers to desktops reporting into Chef Automate.
`Unknown` desktops did not report to Chef Automate during the last twenty-four hours.

### Check-in History

_Check-in History_ shows the history of checked-in desktop counts as graphed over a selected period of time.
Adjust the timescale with the timescale drop-down menu and change the graph display.
The timescale options are "Last 3 Days," "Last 7 Days," and "Last 14 Days".

### Top 10 Errors

The _Top 10 Errors_ display shows the most common errors that have occurred across all desktops over the last twenty-four hours, and a count of the machines that experienced each error.
Selecting an error opens the list view of the desktops (Insight) with the error to enable further investigation.

### Time since Last Check-in

The _Time since Last Check-in_ display shows a count of desktops with an `unknown` status for a defined timefame.
Selecting a time frame row will reveal a summary of associated Desktops.

### Insight (Desktop List)

> Users can click on rows in both _Time since Last Check-in_ and _Top 10 Errors_ to activate this summary list.

Reviewer 2: ```suggestion
Selecting either  _Time since Last Check-in_ or _Top 10 Errors_ activates a summary list.
```

+++
title = "Desktop Dashboard"
description = "Using the Desktop Dashboard"
date = 2018-03-26T16:01:47-07:00
draft = false
bref = ""
toc = true
+++

The Chef Automate _Desktop_ dashboard displays status information about all desktops connected to Chef Automate.
Desktop information populates this dashboard after a Chef Infra Client run has executed.

## Setting Up the Desktop Dashboard

> Enable the Desktop dashboard with: `chef-automate deploy --product automate --product infra-server --product desktop`.

Reviewer: @vjeffrey @susanev Will update and include a link to published Chef Desktop docs after ChefConf.

For more information, see [Install Chef Infra Server with Automate](https://automate.chef.io/docs/infra-server/).
The Desktop dashboard has no supported compliance profiles, and installation with the `--product desktop` flag includes no compliance profiles.

{{< info >}}
When installing Chef Automate with the `--product desktop` flag, _Data Lifecycle_ settings will set by default to not mark nodes as missing and to not delete missing nodes.
We encourage users to not change these specific settings, and not defeat the monitoring purpose of the Desktop dashboard.
{{< /info >}}

## Desktop Dashboard Display

Within Chef Automate, the _Desktop_ dashboard uses four panels to summarize information: _Daily Check-in_ _Check-in History_, _Top 10 Errors_, and _Time since Last Check-in_.

> Selecting rows within these displays will list relevant desktops and selecting an singular desktop reveals its detailed information.

Reviewer: @IanMadd Are you talking about the buttons, not rows, in _Daily Check-in_?

The _Desktop_ dashboard does not support project filtering.
Node counts in the _Desktop_ dashboard may include liveness agents.

### Daily Check-in

The _Daily Check-in_ display shows a top-level view of daily desktop check-in statistics.  
A bar graphic illustrates the proportion of desktops with `unknown` and `checked-in` statuses.
Below the bar, boxes displays counts for desktops with `unknown` and `checked-in` statuses, and a total count of desktops.
`Checked-in` refers to desktops reporting into Chef Automate.
`Unknown` desktops did not report to Chef Automate during the last twenty-four hours.

### Check-in History

_Check-in History_ shows the history of checked-in desktop counts as graphed over a selected period of time.

> Adjust the timescale with the timescale drop-down menu and change the graph display.

Reviewer 5: ```suggestion
Use the drop-down menu to select a different period of time.
```

The timescale options are "Last 3 Days," "Last 7 Days," and "Last 14 Days".

> ### Top 10 Errors

Reviewer 5: Reorder these two sections since "Top 10 Errors" appears after "Time Since Last Check-in" in the UI.

The _Top 10 Errors_ display shows the ten most common errors that have occurred across all desktops over the last twenty-four hours, and a count of the machines that experienced each error.
Selecting an error opens the Filter Desktop List with the applied error filter to enable further investigation.

### Time Since Last Check-in

The _Time since Last Check-in_ display shows a count of desktops with an `unknown` status for defined timefames.
Selecting a row will reveal a Filtered Desktop List of desktops for the selected timeframe.

### Filtered Desktop Lists

> The Filtered Desktop List displays a list of desktops filtered according to the selected timeframe or error in the previous display.
> Selecting rows in either _Time since Last Check-in_ or _Top 10 Errors_ activates this summary list.

Reviewer 5: Maybe put the second line first so it's clear how users will find Filtered Desktop Lists.

```suggestion
Selecting rows in either _Time since Last Check-in_ or _Top 10 Errors_ displays a list of relevant desktops.

The Filtered Desktop List displays a list of desktops filtered according to the selected timeframe or error in the previous display.
```

Apply more filters to this initial list if desired.
Specific filter values populate according to available desktop information and include the categories of "platform," "version," "domain," and "last run status."

> Select an individual desktop row to display the individual desktop's node details.

Reviewer 5: I'd move this line below #### Node Details: 

"Select an individual desktop row to display..."

#### Node Details

The _Node Details_ display shows the details of a single desktop and includes a visualization of its check-in history.

> The individual check-in history displays the desktop's latest status for each day with an associated icon by default.

Reviewer 5: ```suggestion
The individual check-in history displays the desktop's latest status for each day with an associated icon.
```

> Statuses include "converged" (check box icon), "unchanged" (dash icon), "error" (exclamation point icon), or "unknown" (question mark icon).

Reviewer 5: It would be nice if we could include the actual icons in the doc. Maybe not today, but sometime in the future.

Reviewer 3: as a note the icons will change sometime soon

Reviewer: ooooh

Reviewer 3: they are the same general icons, but we are moving from material icons to carbon design system icons. so it will still be a question mark, dash, exclamation point, and question mark. but visually they will be a bit different. so i wouldn't add the actual icons to the doc quite yet.

The check-in history's format can appear in grid icon, status list, or selectable JSON form.
Select the gear icon at the top of the display to make these modifications.
The timescale of the check-in history can also switch between "Last 2 Weeks" and "Last 4 Weeks."

> The individual desktop's information includes a detailed overview, Chef Infra Client, System, Virtualization, and Kernel.

Reviewer 5: ```suggestion
The individual desktop has an information panel that displays the following data: Overview, Chef Infra Client, System, Virtualization, and Kernel.
```

Reviewer: THANK YOU! I was struggling with how to phrase that info.
